recipes/prometheus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class PrometheusCppConan(ConanFile):
+    name = "prometheus-cpp"
+    description = "Prometheus Client Library for Modern C++"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jupp0r/prometheus-cpp"
+    license = "MIT"
+    topics = ("conan", "metrics", "prometheus", "networking")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_pull": [True, False],
+        "with_push": [True, False],
+        "with_compression": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_pull": True,
+        "with_push": True,
+        "with_compression": True,
+    }
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        tools.check_min_cppstd(self, 11)
+
+    def requirements(self):
+        if self.options.with_pull:
+            self.requires("civetweb/1.13")
+        if self.options.with_push:
+            self.requires("libcurl/7.74.0")
+        if self.options.with_compression:
+            self.requires("zlib/1.2.11")
```

Comment:
Compression only exists on the pull https://github.com/jupp0r/prometheus-cpp/search?q=zlib

This should be conditional and not impact package_id when doing push only

https://github.com/jupp0r/prometheus-cpp/blob/bb017ec15a824d3301845a1274b4b46a01d6d871/pull/CMakeLists.txt#L45

Reply:
Good catch. `self.options.with_compression` should be deleted in `configure()` if `not self.options.with_pull`. Afterwards, call `self.options.get_safe("with_compression")`

I would also remove `with_`prefix to all options.